src/managers/TileManager.ts

```diff
@@ -44,46 +46,47 @@ class TileManager {
                 const dto: Nullable<TileDTO> =
                     dtos.find((dto: TileDTO) => dto.id === tile.id) || null;
 
-                let state: TileState = this.identifyTiles(
+                let state: TileState = this.determineState(
                     dto,
-                    session.turnPlayer!,
+                    session.turnPlayer,
                     i,
                 );
                 tile.apply(state, dto);
             }
-            this.getTilesInHand(); //TODO where to save/use
             this.onSync.emit();
         });
     }
 
-    private identifyTiles(
+    private determineState(
         dto: Nullable<TileDTO>,
-        turnPlayer: int,
-        i: int,
+        turnPlayer: Nullable<int>,
+        index: int,
     ): TileState {
+        const InitialAmount: int = this.getInitialAmount(
+            GeneralManager.getPlayers.length,
```

Comment:
getPlayers is a method so this would throw undefinded. you would have to call it, but anyway dont base it on the dtos but on this.list.length.

Reply:
Couldn't you also get the list of players by PlayerManager.getAll() ?

Reply:
Ah yes you are right its about players not tiles, mixed it up. ... to your question: getAll would work most of the time but not in edge cases, because getAll is based on a list which gets created also when generalmanager syncs

Reply:
So in this case you would use getPlayers from the General Manager?